numpy/random/_generator.pyx

```diff
@@ -2997,6 +2998,22 @@ cdef class Generator:
         then the probability distribution of the number of non-"1"s that
         appear before the third "1" is a negative binomial distribution.
 
+        Because this method internally calls ``Generator.poisson`` with an
+        intermediate random value, a ValueError is raised when the choice of 
+        :math:`n` and :math:`p` would result in the mean + 10 sigma of the sampled
+        intermediate distribution exceeding the max acceptable value of the 
+        ``Generator.poisson`` method. This happens when :math:`p` is too low 
+        (a lot of failures happen for every success) and :math:`n` is too big (
+        a lot of sucesses are allowed).
+        Therefore, the :math:`n` and :math:`p` values must satisfy the constraint:
+
+        .. math:: n\\frac{1-p}{p}+10n\\sqrt{n}\\frac{1-p}{p} < 2^{64} - 10\\sqrt{2^{64}},
```

Comment:
One small thing I only thought about - `np.iinfo("i8").max` is actually 2^{63}-1, not 2^{64}.   Sorry about giving you the wrong value earlier.

Reply:
Heh, good catch. Totally missed it when I wrote it. Updated.